app/scaffold-controllers/welcome-screen/welcomeModalController.js

```diff
@@ -0,0 +1,87 @@
+define([
+    'astro-full',
+    'astro-rpc',
+    'bluebird',
+    'plugins/modalViewPlugin',
+    'plugins/secureStorePlugin',
+    'scaffold-controllers/welcome-screen/welcomeController'
+],
+/* eslint-disable */
+function(
+    Astro,
+    AstroRpc,
+    Promise,
+    ModalViewPlugin,
+    SecureStorePlugin,
+    WelcomeController
+) {
+/* eslint-enable */
+
+    var WelcomeModalController = function(modalView) {
+        this.isShowing = false;
+        this.viewPlugin = modalView;
+    };
+
+    WelcomeModalController.init = function() {
+        return Promise.join(
+            ModalViewPlugin.init(),
+            WelcomeController.init(),
+        function(modalView, welcomeController) {
+            modalView.setContentView(welcomeController.viewPlugin);
+
+            // This registers a close handler on the header bar to dismiss
+            // the modal. Without a header bar, the developer is responsible
+            // for implementing a way to dismiss the modal.
+            var welcomeModalController = new WelcomeModalController(modalView);
+            welcomeController.registerCloseEventHandler(function() {
+                welcomeModalController.hide();
+            });
+
+            // Welcome modal RPCs
+            Astro.registerRpcMethod(AstroRpc.names.showWelcome, [], function(res) {
+                welcomeModalController.show({forced: true});
+            });
+
+            Astro.registerRpcMethod(AstroRpc.names.hideWelcome, [], function(res) {
+                welcomeModalController.hide();
+            });
+
+            return welcomeModalController;
+        });
+    };
+
+    WelcomeModalController.prototype.show = function(params) {
+        var self = this;
+        params = Astro.Utils.extend({forced: false}, params);
+
+        var secureStorePromise = SecureStorePlugin.init();
```

Comment:
Would require it being a property of this controller, not sure how we could reference it without it being a property? If so.. How about we initialize it in `app.js` and pass it in to any controller's initialization when needed?